File: backend/LoanThingy.Models2/Loan.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace LoanThingy.Models
{
    public class Loan : ILoan
    {
        public decimal interest { get; set; }
        public decimal averageMonthlyPayment { get; set; }
        public List<Payment> payments { get; set; }
        public int numberOfPayments { get; set; }

        public Loan() { }

        public Loan(decimal interest, decimal averageMonthlyPayment, List<Payment> payments)
        {
            this.interest = interest;
            this.averageMonthlyPayment = averageMonthlyPayment;
            this.payments = payments;
        }

    }
}

File: frontend/src/app/models/Loan.ts
import {Payment} from './Payment';

export class Loan {
    interest: number;
    averageMonthlyPayment: number;
    payments: Payment[];


    constructor(object: any = null) {
        if (object !== null) {
            this.interest = object.interest;
            this.averageMonthlyPayment = object.averageMonthlyPayment;


            this.payments = [];
            if (object.payments !== undefined) {
                for (const item of object.payments) {
                    this.payments.push(new Payment(item));
                }
            }

        }
    }
}
